astropy/coordinates/representation.py

```diff
@@ -1064,10 +1061,17 @@ def _scale_operation(self, op, *args):
         # as operations that returned NotImplemented or a representation
         # instead of a quantity (as would happen for, e.g., rep * rep).
         try:
-            return self.__class__(*results)
+            result = self.__class__(*results)
         except Exception:
             return NotImplemented
 
+        if self.differentials:
```

Comment:
As I noticed when working on ``.transform`` stuff, isn't ``self.differentials`` always a dictionary, even if it's empty?
In which case...
``for key, differential in self.differentials.items():`` will just skip if there are no differentials. The loop doesn't need to be protected.

Reply:
Good point! Fixed.